lint/param-name_checker.go

```diff
@@ -0,0 +1,74 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+)
+
+// ParamNameChecker detects potential issues in function parameter names.
+//
+// Rationale: makes godoc output better.
+type ParamNameChecker struct {
+	ctx *Context
+
+	loudNames map[string]bool
+
+	warnings []Warning
+}
+
+// NewParamNameChecker returns initialized checker for Go functions param names.
+func NewParamNameChecker(ctx *Context) *ParamNameChecker {
+	return &ParamNameChecker{
+		ctx: ctx,
+		loudNames: map[string]bool{
+			"IN":    true,
```

Comment:
Is there any reason to use map instead on set here?

Reply:
There are 2 idioms for "sets" in Go:
- `map[T]bool`
- `map[T]struct{}`
The latter was more efficient in the past, not sure it's accurate anymore.
The fist one is preferred, you can check if element is in set by simply doing `m[key]`.

No "real" sets are builtin or inside stdlib.